packages/integration-tests/src/entities/Image.ts

```diff
@@ -4,6 +4,8 @@ import { Author, Book, ImageCodegen, imageConfig, ImageType, Publisher } from ".
 type ImageOwner = Book | Publisher | Author;
 
 export class Image extends ImageCodegen {
+  // We don't use hasOneThrough or hasOneDerived b/c we use the ImageType to do a
+  // selective .load instead of a load hint that probes every possible table.
```

Comment:
Separately from the comment, I'm wondering if this usage could become:

```
this.hasOneDerived(loadFn, getFn)
```

As an overload to the current:

```
this.hasOneDerived(loadHint, getFn);
```

It should be pretty easy to use TS overloading to support both signatures...

I also haven't added the `this.hasOneDerived` yet. Fwiw I've also tried to add these types of methods to the `BaseEntity` but that also trips up inference, which is weird b/c TS has a `this` type that is specifically supposed to support things like that...